### ISSUE TITLE
loss_weight discrepancy. EDM and AF3

### ISSUE THREAD

**Issue by luwei0917** [lucidrains/alphafold3-pytorch#68]
https://github.com/lucidrains/alphafold3-pytorch/blob/8ef24ebdd69b35874e067e78afa0367b2fcfadef/alphafold3_pytorch/alphafold3.py#L2186

it should be `return (sigma ** 2 + self.sigma_data ** 2) * (sigma + self.sigma_data) ** -2`
follow equation 6 in page 23 of SI. also to have the correct dimension.

**Comment by lucidrains** [comment]
![Screenshot from 2024-07-10 07-11-52](https://github.com/lucidrains/alphafold3-pytorch/assets/108653/dae8056c-0e45-40bc-b24d-9199a8f3baa7)

@luwei0917 i'm following the equation directly from Tero Karras' paper

**Comment by lucidrains** [comment]
@luwei0917 i'm also using code directly from two other repositories using EDM successfully to train image based generative models

**Comment by luwei0917** [comment]
that's a bit strange. because as \sigma become very small, this loss weighting will be very large. but you might be right, as the EDM version could give more weight when t is almost 1.

**Comment by lucidrains** [comment]
@luwei0917 you weren't the only one to point out this discrepancy, we can leave the issue open for further discussion from passerbys

**Comment by lucidrains** [comment]
@luwei0917 hey Wei, i've decided to go with the paper's loss weighting but allow for switching back to Karras' original one with a flag `karras_formulation` on `ElucidatedAtomDiffusion`
